src/presign/record.rs

```diff
@@ -298,7 +292,7 @@ mod tests {
         /// Simulate creation of a random presign record. Do not use outside of
         /// testing.
         fn simulate(rng: &mut StdRng) -> PresignRecord {
-            let mask_point = CurvePoint(ProjectivePoint::random(StdRng::from_seed(rng.gen())));
+            let mask_point = CurvePoint::random(rng.clone());
```

Comment:
In general, we never want to clone RNGs (see e.g. [this issue](https://github.com/boltlabs-inc/tss-ecdsa/pull/472#discussion_r1333566720)).

I would change this function to take a mutable reference and use the `from_seed` method internally if you need an RNG by value.